### PR TITLE
Add C/C++ version

### PR DESCRIPTION
**Pull request by giusbianco**
Added C and C++ versions of `libfakedevice` using meson as the building system. Added `build.yml` and `deploy.yml`.  The first file defines a build pipeline for the three projects, while the second file defines a deploy pipeline.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install meson clang
+        sudo meson wrap update-db
```
Comment (Luni-4) [inline .github/workflows/build.yml]: Is this line a leftover or is it necessary?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            - name: Create a release
+              uses: softprops/action-gh-release@v1
+              with:
+                tag_name: v1.0.0 
+                files: |
+                    ./binaries/fake-firmware-cpp
+                    ./binaries/fake-firmware-c
+                    ./binaries/fake-firmware
+              env:
+                GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Luni-4) [inline .github/workflows/deploy.yml]: Let's remove this space at the end of the script

File: libfakedevice-c/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#### `accessAudioDriver()`
+- Accesses the audio driver and produces a two-seconds beep sound.
+
+### 3. Fake Firmware (`fake-firmware.c`)
+- Integrates and utilizes functionalities from both `device` and `feature`.
+- Demonstrates the combined usage of features in a simulated firmware.
+
+## Dependencies
+
+Before building and running the project, make sure that `Meson` and `CLang` are installed on your system. In particular, `Meson` (https://mesonbuild.com/) is used as a tool to build the project, while `CLang` (https://clang.llvm.org/) is used as a compiler for the project. We also suggest updating the `wrapdb` (https://mesonbuild.com/Wrap-dependency-system-manual.html) used by Meson to retrieve the dependencies presented in the next section.
```
Comment (Luni-4) [inline libfakedevice-c/README.md]: `CLang` or `Clang`?

File: libfakedevice-c/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+### Other dependencies
+
+The project relies on the following external libraries:
```
Comment (Luni-4) [inline libfakedevice-c/README.md]: Let's specify that we are saying how to install these dependencies on `Linux`

File: libfakedevice-c/src/fake-firmware.c [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    // Access audio driver
+    accessAudioDriver();
+    
+    // Turn a lamp on   
+    turnLampOn();
+    // Turn lamp off
+    turnLampOff();
+
+    return 0;
+}
```
Comment (Luni-4) [inline libfakedevice-c/src/fake-firmware.c]: Let's remove the space at the end of this file

File: libfakedevice-c/include/device.h [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#ifndef DEVICE_H
+#define DEVICE_H
+
+#include <stdbool.h>
+
+bool turnLampOn();
+bool turnLampOff();
+
+#endif // DEVICE_H
```
Comment (Luni-4) [inline libfakedevice-c/include/device.h]: Let's remove the space at the end of this file

File: libfakedevice-cpp/README.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+2. [rtaudio](https://github.com/thestk/rtaudio): Used for audio generation. Install it using:
+```bash
+sudo apt update
+sudo apt install librtaudio-dev
+```
+
+## Build and Run the project
+
+The complete compilation procedure can be started by running `./autorun.sh`. You can launch the project by running `./libfakedevice_cpp`.
```
Comment (Luni-4) [inline libfakedevice-cpp/README.md]: `launch the produced binary`

Same comments for the `C`'s README

File: libfakedevice-cpp/src/feature.cpp [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    for (int i = 0; i < 10; ++i) {
+        ssize_t bytesRead = read(fd, buffer, buffer_size);
+        if (bytesRead == -1) {
+            // Handle error if reading data fails
+            cerr << "Error reading data" << endl;
+            close(fd);
+            return;
+        }
+    }
+    cout << "Nice video!" << endl;
```
Comment (Luni-4) [inline libfakedevice-cpp/src/feature.cpp]: Can we change this in a more appropriate message? It's funny as it was written though :smile:

File: libfakedevice-c/src/feature.c [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+    for (int i = 0; i < 10; ++i) {
+        ssize_t bytesRead = read(fd, buffer, buffer_size);
+        if (bytesRead == -1) {
+            // Handle error if reading data fails
+            fprintf(stderr, "Error reading data\n");
+            close(fd);
+            return;
+        }
+    }
+    printf("Nice video!\n");
```
Comment (Luni-4) [inline libfakedevice-c/src/feature.c]: Change the message here too

File: libfakedevice-c/README.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+2. [openal](https://github.com/kcat/openal-soft): Used for audio generation. Install it using:
+```bash
+sudo apt update
+sudo apt install libopenal-dev
+```
+
+## Build and Run the project
+
+The complete compilation procedure can be started by running `./autorun.sh`. You can launch the project by running `./libfakedevice_c`.
```
Comment (Luni-4) [inline libfakedevice-c/README.md]: Same comment for the `C++`

**Review by Luni-4 — CHANGES_REQUESTED** [review]
Thanks a lot for your hard work! Just simple suggestions

File: libfakedevice-cpp/src/feature.cpp (hunk 8 above)
Comment (giusbianco) [reply libfakedevice-cpp/src/feature.cpp]: oh no, it's embarrassing! I forgot to remove it!

File: .github/workflows/build.yml (hunk 1 above)
Comment (giusbianco) [reply .github/workflows/build.yml]: Strange, I had removed it! Anyway, it is not necessary.

**Review by Luni-4 — APPROVED** [review]

**Pull request merged**